Client/Coordinators/Router/RouteBuilder.swift

```diff
@@ -135,4 +135,25 @@ struct RouteBuilder {
             TelemetryWrapper.recordEvent(category: .action, method: .open, object: .largeTabsOpenUrl)
         }
     }
+
+    func makeRoute(shortcutItem: UIApplicationShortcutItem) -> Route? {
+        guard let shortcutTypeRaw = shortcutItem.type.components(separatedBy: ".").last,
+              let shortcutType = DeeplinkInput.Shortcut(rawValue: shortcutTypeRaw)
+        else { return nil }
+
+        switch shortcutType {
+        case .newTab:
+            return .search(url: nil, isPrivate: false)
+        case .newPrivateTab:
+            return .search(url: nil, isPrivate: true)
+        case .openLastBookmark:
+            if let urlToOpen = (shortcutItem.userInfo?[QuickActionInfos.tabURLKey] as? String)?.asURL {
+                return .search(url: urlToOpen, isPrivate: false)
```

Comment:
The current code also makes sure to `switchToPrivacyMode` to false before switching to a tab with a particular URL or creating that tab (with the switchToTabForURLOrOpen method). We should keep that behavior.